### PR TITLE
fix eventmap handling

### PR DESCRIPTION
**Pull request by Ankurk99**

### CODE REVIEW

File: KubeArmor/BPF/entrypoint.bpf.c [hunk 1]
```diff
@@ -29,9 +29,12 @@ enum {
 
 int skip_syscall(uint32_t key)
 {
-    uint32_t *val = bpf_map_lookup_elem(&ka_ea_event_map, &key);
-    if (val == 0) return 1;
-    else if (val != 0 && *val == 0) return 1;
+    uint32_t *val;
+
+    val = bpf_map_lookup_elem(&ka_ea_event_map, &key);
+    if (!val || *val == 0)
```
Comment (nam-jaehyun) [inline KubeArmor/BPF/entrypoint.bpf.c:35]: why did you remove the line to check if the 'val' pointer is null?

File: KubeArmor/eventAuditor/entryPointHandler.go [hunk 2]
```diff
@@ -132,30 +133,44 @@ func (ea *EventAuditor) DestroyEntryPointPrograms(bman *KABPFManager) error {
 func (ea *EventAuditor) EnableEntryPoint(probe string) {
 	var eventMapElem EventElement
 
+	probe = strings.Replace(probe, "sys_", "", -1)
```
Comment (nam-jaehyun) [inline KubeArmor/eventAuditor/entryPointHandler.go]: if strings.HasPrefix(probe, "sys_") {
   ...
}
we need to make a room for other kprobes too.

File: KubeArmor/eventAuditor/entryPointHandler.go [hunk 3]
```diff
@@ -132,30 +133,44 @@ func (ea *EventAuditor) DestroyEntryPointPrograms(bman *KABPFManager) error {
...
+	_, supported := ea.SupportedEntryPoints[probe]
+
+	if !supported {
+		fmt.Println(probe, "is currently not supported")
```
Comment (nam-jaehyun) [inline KubeArmor/eventAuditor/entryPointHandler.go]: ea.Logger.Errf("%s is currently not supported", probe)

File: KubeArmor/eventAuditor/entryPointHandler.go [hunk 4]
```diff
@@ -132,30 +133,44 @@ func (ea *EventAuditor) DestroyEntryPointPrograms(bman *KABPFManager) error {
...
 	eventMapElem.SetKey(uint32(ea.SupportedEntryPoints[probe]))
 	eventMapElem.SetValue(uint32(1))
 
-	// TODO: value is not updated
 	if err := ea.BPFManager.MapUpdateElement(&eventMapElem); err != nil {
 		ea.Logger.Errf("Failed to update KAEAEventMap (attachEntryPoint, %s, %d) (%v)", probe, err)
 	}
 
-	// TODO: remove it once fixing the update
+	// for debugging
```
Comment (nam-jaehyun) [inline KubeArmor/eventAuditor/entryPointHandler.go]: If the logic works, we can remove 151-152.

File: KubeArmor/eventAuditor/entryPointHandler.go [hunk 5]
```diff
@@ -132,30 +133,44 @@ func (ea *EventAuditor) DestroyEntryPointPrograms(bman *KABPFManager) error {
...
 	fmt.Println("EnableEntryPoint", eventMapElem)
 }
 
 func (ea *EventAuditor) DisableEntryPoint(probe string) {
 	var eventMapElem EventElement
 
+	probe = strings.Replace(probe, "sys_", "", -1)
```
Comment (nam-jaehyun) [inline KubeArmor/eventAuditor/entryPointHandler.go]: the same comment with the above one.

File: KubeArmor/eventAuditor/entryPointHandler.go [hunk 6]
```diff
@@ -132,30 +133,44 @@ func (ea *EventAuditor) DestroyEntryPointPrograms(bman *KABPFManager) error {
...
+	_, supported := ea.SupportedEntryPoints[probe]
+
+	if !supported {
+		fmt.Println(probe, "is currently not supported")
```
Comment (nam-jaehyun) [inline KubeArmor/eventAuditor/entryPointHandler.go]: the same comment with the above one.

File: KubeArmor/eventAuditor/entryPointHandler.go [hunk 7]
```diff
@@ -132,30 +133,44 @@ func (ea *EventAuditor) DestroyEntryPointPrograms(bman *KABPFManager) error {
...
 	eventMapElem.SetKey(uint32(ea.SupportedEntryPoints[probe]))
 	eventMapElem.SetValue(uint32(0))
 
-	// TODO: value is not updated
 	if err := ea.BPFManager.MapUpdateElement(&eventMapElem); err != nil {
 		ea.Logger.Errf("Failed to update KAEAEventMap (DetachEntryPoint, %s, %d) (%v)", probe, err)
 	}
 
-	// TODO: remove it once fixing the update
+	// for debugging
```
Comment (nam-jaehyun) [inline KubeArmor/eventAuditor/entryPointHandler.go]: the same comment with the above one.

File: KubeArmor/eventAuditor/entryPointHandler.go [hunk 8]
```diff
@@ -172,6 +187,7 @@ func (ea *EventAuditor) UpdateEntryPoints(auditPolicies *map[string]tp.AuditPoli
 	// all entrypoints list
 	for _, policy := range AuditPolicies {
 		for _, event := range policy.Events {
+			event.Probe = strings.Replace(event.Probe, "sys_", "", -1)
```
Comment (nam-jaehyun) [inline KubeArmor/eventAuditor/entryPointHandler.go]: the same comment with the above one.

File: KubeArmor/eventAuditor/entryPointHandler.go [hunk 9]
```diff
@@ -182,6 +198,7 @@ func (ea *EventAuditor) UpdateEntryPoints(auditPolicies *map[string]tp.AuditPoli
 
 	// new entrypoints to be attached
 	for _, newProbe := range entrypointList {
+		newProbe = strings.Replace(newProbe, "sys_", "", -1)
```
Comment (nam-jaehyun) [inline KubeArmor/eventAuditor/entryPointHandler.go]: the same comment with the above one.

File: KubeArmor/eventAuditor/entryPointHandler.go [hunk 10]
```diff
@@ -196,6 +213,7 @@ func (ea *EventAuditor) UpdateEntryPoints(auditPolicies *map[string]tp.AuditPoli
 
 	// old entrypoints to be detached
 	for _, oldProbe := range ea.ActiveEntryPoints {
+		oldProbe = strings.Replace(oldProbe, "sys_", "", -1)
```
Comment (nam-jaehyun) [inline KubeArmor/eventAuditor/entryPointHandler.go]: the same comment with the above one.

File: KubeArmor/BPF/entrypoint.bpf.c (hunk 1 above)
Comment (Ankurk99) [reply KubeArmor/BPF/entrypoint.bpf.c:35]: Here !val is checking for the previous (val == 0) condition

File: KubeArmor/eventAuditor/entryPointHandler.go (hunk 2 above)
Comment (Ankurk99) [reply KubeArmor/eventAuditor/entryPointHandler.go]: Okay, will change it

File: KubeArmor/eventAuditor/entryPointHandler.go (hunk 4 above)
Comment (Ankurk99) [reply KubeArmor/eventAuditor/entryPointHandler.go]: We are not getting the required output from the trace_pipe, even when the map is being correctly updated. This part was left for debugging this error.

Comment (nam-jaehyun) [reply KubeArmor/eventAuditor/entryPointHandler.go]: you mean the current code is not working?

Comment (Ankurk99) [reply KubeArmor/eventAuditor/entryPointHandler.go]: yes, there is some issue with the "val" in entrypoint.bpf.c, we are not getting the required output from the trace_pipe.
The syscalls whose value is set to 1 (and are enabled), are being skipped.
